java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -64,25 +65,35 @@ public void visitNode(Tree tree) {
       return;
     }
 
+    boolean springInjectionAnnotated = isSpringInjectionAnnotated(methodSymbol.metadata());
     methodTree.parameters().stream()
       .map(VariableTree::symbol)
       .filter(p -> p.type().isClass() && !p.type().symbol().isEnum() && !p.type().is("java.lang.String"))
-      .forEach(this::handleParameter);
+      .forEach(p -> handleParameter(p, springInjectionAnnotated));
   }
 
   private static boolean isOverloaded(Symbol.MethodSymbol methodSymbol) {
     return ((Symbol.TypeSymbol) methodSymbol.owner()).lookupSymbols(methodSymbol.name()).size() > 1;
   }
 
-  private void handleParameter(Symbol parameter) {
+  private void handleParameter(Symbol parameter, boolean springInjectionAnnotated) {
     Type leastSpecificType = findLeastSpecificType(parameter);
     if (parameter.type() != leastSpecificType
-      && !leastSpecificType.is("java.lang.Object")) {
-      String suggestedType = leastSpecificType.fullyQualifiedName().replace('$', '.');
+      && !leastSpecificType.is("java.lang.Object")
+      && !(springInjectionAnnotated && parameter.type().is("java.util.Collection"))) {
```

Comment:
matter of style and choices but I tend to think that this last part of the &&  should be implemented as a filter of stream on the parameters stream.